### DISCUSSION TITLE
Worker Processes initiates background scheduler everytime.

### DISCUSSION THREAD

**Discussion by debadatta-tiger-analytics** [fastapi/fastapi#10603] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.on_event('startup')
    def schedule_job():
         """
         logger.info('*********Execute********')
         scheduler = BackgroundScheduler()
         logger.info('*********job1 fn called********')
         scheduler.add_job(job1, trigger='cron', minute=os.getenv('SYNC_DB'))
         logger.info('*********job2 fn called********')
         scheduler.add_job(job2, trigger='cron',minute=os.getenv('SCHEDULE_POPULATE_EXTERNAL_DB'))
         logger.info('*********job3 fn called********')
         scheduler.add_job(job3 ,CronTrigger.from_crontab(os.getenv('NOTIFICATION_SCHEDULE_TIME')), 
             max_instances=1)
         scheduler.start()
```

### Description

This is my on startup event, and I have 3 background processes.
   
   
   The command I am using to run my FastAPI app is -> 
   `gunicorn -w 4 -k uvicorn.workers.UvicornWorker  app.main:app --preload --timeout 30000`
   
   So, due to 4 workers, the jobs are getting called 4 times. How to avoid this?
   
   
   I have tried having 3 individual startup functions, that is also not working, one of the job is to send mails periodically, so it sends same mail 4 times to a user

### Operating System

Windows

### FastAPI Version

0.89.1

### Pydantic Version

1.10.6

### Python Version

3.11.4

**Comment by jgould22** [comment, marked as answer]
Often this type of functionality is abstracted outside of the application using something like celery.

You may be able to achieve something similar by moving this functionality into Gunicorns start up hook https://docs.gunicorn.org/en/stable/settings.html#server-hooks
